rig/machine_control/boot.py

```diff
@@ -0,0 +1,160 @@
+
+spin1_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 0,
+    "led_config": 0x00076104,
+}
+spin2_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 2,
+    "led_config": 0x00006103,
+}
+spin3_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 3,
+    "led_config": 0x00000502,
+}
+spin4_boot_options = {
+    "width": 8, "height": 8, "hardware_version": 4,
+    "led_config": 0x00000001,
+}
+spin5_boot_options = {
+    "width": 8, "height": 8, "hardware_version": 5,
+    "led_config": 0x00000001,
+}
+
+
+def boot(hostname, width, height, cpu_frequency=200, hardware_version=0,
+         led_config=0x00000001, boot_data=None, struct_data=None):
+    """Boot a SpiNNaker machine of the given size.
+
+    Parameters
+    ----------
+    hostname : str
+        Hostname or IP address of the SpiNNaker board to boot.
+    width : int
+        Width of the machine (0 < w < 256)
+    height : int
+        Height of the machine (0 < h < 256)
+    cpu_frequency : int
+        CPU clock-frequency.  **Note**: The default (200 MHz) is known
+        safe.
+    hardware_version : int
+    led_config : int
+    boot_data : bytes
+        Data to boot the machine with.
+    struct_data : bytes or None
+        Data to interpret as a representation of the memory layout of data
+        within the boot data.
+
+    Notes
+    -----
+    The constants `rig.boot.spinX_boot_options` can be used to specify boot
+    parameters, for example:
+
+        boot("board1", **spin5_boot_options)
+
+    Will boot the Spin5 board connected with hostname "board1".
+    """
+    # Read the struct file and modify the "sv" struct to contain the
+    # configuration values and write this into the boot data.
+    sv = struct_file.read_struct_file(struct_data)[b"sv"]
+    sv.update_default_values(p2p_dims=(width << 8) | height,
+                             hw_ver=hardware_version,
+                             cpu_clk=cpu_frequency,
+                             led0=led_config,
+                             unix_time=int(time.time()),
+                             boot_sig=int(time.time()),
+                             root_chip=1)
+    struct_packed = sv.pack()
+
+    # NOTE I am assured by ST that writing into boot data bytes 384 onwards
+    # is unlikely to change in the foreseeable future.
+    buf = bytearray(boot_data)
+    buf[BOOT_DATA_OFFSET:BOOT_DATA_OFFSET+BOOT_DATA_LENGTH] = \
+        struct_packed[:BOOT_DATA_LENGTH]
+    assert len(buf) < 32768  # Assert that we fit in DTCM
+    boot_data = bytes(buf)
+
+    # Create a socket to communicate with the board
+    sock = socket.socket(type=socket.SOCK_DGRAM)
+    sock.connect((hostname, PORT))
+
+    # Transmit the boot data as a series of SDP packets.  First determine
+    # how many blocks must be sent and transmit that, then transmit each
+    # block.
+    n_blocks = len(buf) // BOOT_BYTE_SIZE + (
+        1 if len(buf) % BOOT_BYTE_SIZE != 0 else 0)
```

Comment:
`(len(buf) + BOOT_BYTE_SIZE - 1) // BOOT_BYTE_SIZE`

Isn't that the more conventional way to get the ceiling truncation?